actions/checkForAndPlaceOrder.ts

```diff
@@ -415,15 +415,58 @@ function _handleGetTradableOrderCall(
   if (error.code === Logger.errors.CALL_EXCEPTION) {
     const errorMessagePrefix =
       "[getTradeableOrderWithSignature] Call Exception";
-    switch (error.errorName) {
+
+    // TODO: When Nethermind hopefully uses the same error format as other nodes, we can remove this hack
+    //
+    // Nethermind does some weird stuff with the error message, so we need to do some magic 💫
+    // Assumptions:
+    // - `error.data` exists for all tested RPC nodes.
+    // - All revert reasons in the code base are expected to return a non-zero `error.data`
+    // - Nethermind, irrespective of the revert reason, returns a zero-bytes `error.data`
+    // Therefore:
+    // - Nethermind: `error.data` in a revert case is `0x` (empty string), with the revert reason buried in
+    //               `error.error.error.data`
+    // - Other nodes: `error.data` in a revert case we expect the revert reason / custom error selector
+    // Therefore, if it's a Nethermind reversion, it takes the format `Reverted 0x01234567` where `01234567` is
+    // the error selector. We can therefore check if the `error.data` is `0x` and if so, we can extract the
+    // error selector from `error.error.error.data` and use that to determine the error.
+    let parsedError = "";
+    if (error.data === "0x") {
+      // Verify that `error.error.error.data` is defined and is a string, otherwise it
+      // could be a different error
+      // TODO: Is there a better way to do this?
+      if (
+        error.error &&
+        error.error.error &&
+        typeof error.error.error.data === "string"
+      ) {
+        // 9 skips the `Reverted ` prefix, and 19 is the end of the error selector
+        parsedError = String(error.error.error.data).slice(9, 19);
+      }
+    } else {
+      parsedError = error.errorName; // If it's not nethermind, we can use the error name
+    }
+
+    switch (parsedError) {
       case "OrderNotValid":
+      case "0xc8fc2725":
         // The conditional order has not expired, or been cancelled, but the order is not valid
         // For example, with TWAPs, this may be after `span` seconds have passed in the epoch.
+
+        // As the `OrderNotValid` is parameterized, we can retrieve the reason. This is an abi encoded string,
+        // so we need to decode it.
+        // TODO: use the `OrderNotValid` reason?
+        const reason = ethers.utils.defaultAbiCoder.decode(
+          ["string"],
+          String(error.error.error.data).slice(19)
```

Comment:
I feel the `error.error.error.data`, which seems prone to error, is being processed in more than one branch. In one of the branches, you had an if in case `error.error.error.data` was not defined, but not in this one.

Maybe, before the switch we can just process the error and extract the relevant info, and then we enter the switch with clean data

Reply:
This has been refined so that their magic is only handled in one branch, with heavy validation of the assumptions.